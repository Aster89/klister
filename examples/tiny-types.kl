#lang kernel

(example (the (Bool) (true)))
(example (the (Bool) (false)))
(example (the (Syntax) 'foo))
(example (the (Macro (Syntax)) (pure 'foo)))
(example (the (-> (Bool) (Bool)) (lambda (x) x)))
(example (the (-> (Bool) (Syntax)) (lambda (x) 'hello)))
(example (the (Macro (-> (Bool) (Bool)))
              (>>= (free-identifier=? 'a 'b)
                   (lambda (eq?)
                     (case eq?
                      [(true) (pure (lambda (x) x))]
                      [(false) (pure (lambda (x)
                                       (case x
                                         [(true) (false)]
                                         [(false) (true)])))])))))
                       





