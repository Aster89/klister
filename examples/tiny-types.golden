(true) : Bool
(false) : Bool
#[tiny-types.kl:5.25-5.28]<foo> : Syntax
pure #[tiny-types.kl:6.39-6.42]<foo> : (Macro Syntax)
#<closure> : (Bool → Bool)
#<closure> : (Bool → Syntax)
(free-identifier=?
 #[tiny-types.kl:10.40-10.41]<a>
 #[tiny-types.kl:10.43-10.44]<b>)
>>=
#<closure> : (Macro (Bool → Bool))
